### ISSUE TITLE
double tap zoom on customChild

### ISSUE THREAD

**Issue by clragon** [fireslime/photo_view#342]
**Is your feature request related to a problem? Please describe.**
Using `PhotoViewGalleryPageOptions.customChild` instead of `PhotoViewGalleryPageOptions` will result in the double tap gesture losing its function.
If you zoom in manually, double tapping will still unzoom the child, 
however, zooming in cannot be done by double tapping on a custom child.

**Describe the solution you'd like**
I would like to be able to trigger the next scale state with a double tap on a custom child.
Specifically, my code looks something like this:
```dart
return PhotoViewGallery.builder(
  builder: (context, index) {
      return PhotoViewGalleryPageOptions.customChild(
      child: CachedNetworkImage(
          imageUrl: posts[index].highres,
          placeholder: (context, url) => CachedNetworkImage(
              imageUrl: posts[index].lowres,
              placeholder: (context, url) => CircularProgressIndicator(),
          ),
      ),
      )
  },
)
```

**Describe alternatives you've considered**
I tried using nested photoviews like this:
```dart
return PhotoViewGallery.builder(
  builder: (context, index) {
    return PhotoViewGalleryPageOptions.customChild(
      child: PhotoView(
        backgroundDecoration: BoxDecoration(
          color: Colors.transparent,
        ),
        imageProvider: CachedNetworkImageProvider(
            imageUrl: posts[index].highres),
        loadingBuilder: (context, chunk) => PhotoView(
          imageProvider: CachedNetworkImageProvider(
              imageUrl: posts[index].lowres),
        ),
      ),
  },
);
```
but it will result in a mess. 

I tried implementing this myself by wrapping the gallery inside a gesturedetector, but this is not possible.

**Comment by clragon** [comment]
Alright after a bit of trying things out I have found that the following works:

```dart
return PhotoViewGallery.builder(
  builder: (context, index) {
    return PhotoViewGalleryPageOptions.customChild(
      child: PhotoView(
        backgroundDecoration: BoxDecoration(
          color: Colors.transparent,
        ),
        imageProvider: CachedNetworkImageProvider(
            widget.posts[index].image.value.file['url']),
        loadingBuilder: (context, chunk) => PhotoView(
          imageProvider: CachedNetworkImageProvider(
              widget.posts[index].image.value.sample['url']),
        ),
      ),
  },
);
```

Which will provide the double tap zoom, because it seems the double tap zoom is actually part of the `PhotoView` and is not applied if youre using a custom child _at all_, not even a `PhotoView.customChild`.
The only problem with this solution is that when the full screen image is loaded, the zoom will reset.

**Comment by clragon** [comment]
Alright I have tried some more things, but I am just shifting the issue around.
this will restore double tap:
```dart
return PhotoViewGallery.builder(
  builder: (context, index) {
      return PhotoViewGalleryPageOptions.customChild(
      childSize: Size(posts[index].highres.width, posts[index].highres.height)
      child: CachedNetworkImage(
          imageUrl: posts[index].highres,
          placeholder: (context, url) => CachedNetworkImage(
              imageUrl: posts[index].lowres,
              placeholder: (context, url) => CircularProgressIndicator(),
          ),
      ),
      )
  },
)
```
but during loading, the low res picture will be displayed very small.

I assume rebuilding the gallery with state will reset the zoom so I am where I was before.

**Comment by clragon** [comment]
I got it.

Using the lowres version as child size works.
The low res version will be fitted from the start and somehow this will still work fine with the high res version.
maybe the zoom isnt as strong when the actual high res version is loaded.
anyways, this works very well so far, Imma close this issue for now.

```dart
return PhotoViewGallery.builder(
  builder: (context, index) {
      return PhotoViewGalleryPageOptions.customChild(
      childSize: Size(posts[index].lowres.width, posts[index].lowres.height)
      child: CachedNetworkImage(
          imageUrl: posts[index].highres,
          placeholder: (context, url) => CachedNetworkImage(
              imageUrl: posts[index].lowres,
              placeholder: (context, url) => CircularProgressIndicator(),
          ),
      ),
      )
  },
)

```

**Comment by cherrybiu** [comment]
> I got it.
> 
> Using the lowres version as child size works.
> The low res version will be fitted from the start and somehow this will still work fine with the high res version.
> maybe the zoom isnt as strong when the actual high res version is loaded.
> anyways, this works very well so far, Imma close this issue for now.
> 
> ```dart
> return PhotoViewGallery.builder(
>   builder: (context, index) {
>       return PhotoViewGalleryPageOptions.customChild(
>       childSize: Size(posts[index].lowres.width, posts[index].lowres.height)
>       child: CachedNetworkImage(
>           imageUrl: posts[index].highres,
>           placeholder: (context, url) => CachedNetworkImage(
>               imageUrl: posts[index].lowres,
>               placeholder: (context, url) => CircularProgressIndicator(),
>           ),
>       ),
>       )
>   },
> )
> ```

I can understand you ,but what confused me was the String has no 'width' property. it seems that the `posts[index].lowres` is String. Could you please show me the `posts[index]`

**Comment by clragon** [comment]
Thats because this is pseudo code. My actual code does not look like this.
You will have to replace `posts[index].lowres.width` with whatever gives you the width of your image, same for height. 
In my Code I have an object that I call with
`widget.posts['index'].image` and I get `.file` for highres and `.sample` for lowres.
Then in the imageUrl I pass `.image.file.url` and `.image.file.sample`,
in the `Size` I pass `.image.sample.width`, `.image.sample.height`

**Comment by cherrybiu** [comment]
Thanks , I get it. And it did work~

**Comment by clragon** [comment]
My journey continues, because I have found several issues.

First of all, the loading builder of the cached network image will be treated as part of the image, which results in the widgets being really small or big, dependent in the image itself. I tried fixing that, but I have no clue how I would approach that. 

The second issue is that using the lower res version's height and width as the page height and widht will nullify the second zoom step in most cases. This means I would need to update the height and width when the larger image is loaded.
I tried doing that, but its not really working out well. Progress is tracked over at https://github.com/Baseflow/flutter_cached_network_image/issues/449
because I feel like I might be able to hack through this with an onLoad call. 

I'm writing this just in case someone else has a similar problem.
Maybe it would be easier to just fork this repository here and implement some custom logic that that allows me to use a correctly sized loading builder and a lower as well as a higher res image widget that.
